packages/app/src/app.ts

```diff
@@ -387,17 +391,30 @@ export class App<Req extends Request = Request, Res extends Response = Response>
 
     let idx = 0
 
-    const loop = () => res.writableEnded || (idx < mw.length && handle(mw[idx++])(req, res, next))
```

Comment:
a function rename is unnecessary

Reply:
Happy to revert, but maybe we can agree on a more suitable name? Both 'loop' and 'handleNext' are not great at explaining what it does.

Reply:
I don't know about a more suitable name but renaming stuff isn't in the scope of this PR. The PR is fixing a bug.

Reply:
That logic is really flawed - in order to fix a bug, often you have to refactor things. Trying to keep names the same whilst changing behaviour means intent/behaviour will diverge from the variable/function names, reducing code quality.

You've expressed you aren't interested in trying to agree on a better name, so I'll revert to `loop`

Reply:
it's not flawed, the function name "loop" corresponds to middleware dispatch loop. That's why it's called as such. It was taken from Polka's source where it does the same thing and is called the same

Reply:
No, that's not what I mean. I see why it's called `loop` and I think that logic is sound (though I don't think the name is perfect).

The logic of 'this PR is a fix, so we can't rename anything' is what I am referring to.